analyses/snv-callers/scripts/01-setup_db.py

```diff
@@ -0,0 +1,282 @@
+    ('CCDS', 'TEXT'),
+    ('ENSP', 'TEXT'),
+    ('SWISSPROT', 'TEXT'),
+    ('TREMBL', 'TEXT'),
+    ('UNIPARC', 'TEXT'),
+    ('RefSeq', 'TEXT'),
+    ('SIFT', 'TEXT'),
+    ('PolyPhen', 'TEXT'),
+    ('EXON', 'TEXT'),
+    ('INTRON', 'TEXT'),
+    ('DOMAINS', 'TEXT'),
+    ('AF', 'TEXT'),
+    ('AFR_AF', 'TEXT'),
+    ('AMR_AF', 'TEXT'),
+    ('ASN_AF', 'TEXT'),
+    ('EAS_AF', 'TEXT'),
+    ('EUR_AF', 'TEXT'),
+    ('SAS_AF', 'TEXT'),
+    ('AA_AF', 'TEXT'),
+    ('EA_AF', 'TEXT'),
+    ('CLIN_SIG', 'TEXT'),
+    ('SOMATIC', 'TEXT'),
+    ('PUBMED', 'TEXT'),
+    ('MOTIF_NAME', 'TEXT'),
+    ('MOTIF_POS', 'INTEGER'),
+    ('HIGH_INF_POS', 'TEXT'),
+    ('MOTIF_SCORE_CHANGE', 'REAL'),
+    ('IMPACT', 'TEXT'),
+    ('PICK', 'TEXT'),
+    ('VARIANT_CLASS', 'TEXT'),
+    ('TSL', 'TEXT'),
+    ('HGVS_OFFSET', 'TEXT'),
+    ('PHENO', 'TEXT'),
+    ('MINIMISED', 'TEXT'),
+    ('ExAC_AF', 'TEXT'),
+    ('ExAC_AF_AFR', 'TEXT'),
+    ('ExAC_AF_AMR', 'TEXT'),
+    ('ExAC_AF_EAS', 'TEXT'),
+    ('ExAC_AF_FIN', 'TEXT'),
+    ('ExAC_AF_NFE', 'TEXT'),
+    ('ExAC_AF_OTH', 'TEXT'),
+    ('ExAC_AF_SAS', 'TEXT'),
+    ('GENE_PHENO', 'TEXT'),
+    ('FILTER', 'TEXT'),
+    ('flanking_bps', 'TEXT'),
+    ('vcf_id', 'TEXT'),
+    ('vcf_qual', 'REAL'),
+    ('ExAC_AF_Adj', 'TEXT'),
+    ('ExAC_AC_AN_Adj', 'TEXT'),
+    ('ExAC_AC_AN', 'TEXT'),
+    ('ExAC_AC_AN_AFR', 'TEXT'),
+    ('ExAC_AC_AN_AMR', 'TEXT'),
+    ('ExAC_AC_AN_EAS', 'TEXT'),
+    ('ExAC_AC_AN_FIN', 'TEXT'),
+    ('ExAC_AC_AN_NFE', 'TEXT'),
+    ('ExAC_AC_AN_OTH', 'TEXT'),
+    ('ExAC_AC_AN_SAS', 'TEXT'),
+    ('ExAC_FILTER', 'TEXT'),
+    ('gnomAD_AF', 'TEXT'),
+    ('gnomAD_AFR_AF', 'TEXT'),
+    ('gnomAD_AMR_AF', 'TEXT'),
+    ('gnomAD_ASJ_AF', 'TEXT'),
+    ('gnomAD_EAS_AF', 'TEXT'),
+    ('gnomAD_FIN_AF', 'TEXT'),
+    ('gnomAD_NFE_AF', 'TEXT'),
+    ('gnomAD_OTH_AF', 'TEXT'),
+    ('gnomAD_SAS_AF', 'TEXT'),
+    ('vcf_pos', 'INTEGER'),
+    ('VAF', 'REAL')
+]
+
+needed_cols = [
+    'Hugo_Symbol',
+    'Entrez_Gene_Id',
+    'Chromosome',
+    'Start_Position',
+    'End_Position',
+    'Variant_Classification',
+    'Variant_Type',
+    'Reference_Allele',
+    'Tumor_Seq_Allele1',
+    'Tumor_Seq_Allele2',
+    'dbSNP_RS',
+    'Tumor_Sample_Barcode',
+    'Transcript_ID',
+    't_depth',
+    't_ref_count',
+    't_alt_count',
+    'n_depth',
+    'n_ref_count',
+    'n_alt_count',
+    'Allele',
+    'flanking_bps',
+    'VAF'
+]
+
+needed_types = [col for col in maf_types if col[0] in needed_cols]
+
+# indexes to create. lists to handle multiple column indexes.
+maf_indexes = [
+    ('SNV', ['Chromosome', 'Start_Position', 'Reference_Allele', 'Allele']),
+    ('sample', ['Tumor_Sample_Barcode'])
+]
+
+# translate sqltypes to dtypes
+dtypes = {
+    'TEXT': 'object',
+    'INTEGER': 'Int64',
+    'REAL': 'float64'
+}
+maf_dtypes = {col: dtypes[sql_type] for col, sql_type in maf_types}
+
+# connect to/create the db
+con = sqlite3.connect(args.db_file)
+
+chunksize = 1e5
```

Comment:
Is there any reason this should be a parse option?

Reply:
Arguably yes, but this seemed a good default, and it makes almost no speed difference I could discern for reasonable values.